src/SFA.DAS.UnitTests/ExecutionPolicies/HmrcExecutionPolicyTests/PolicyRetryTest.cs

```diff
@@ -0,0 +1,94 @@
+﻿using HMRC.ESFA.Levy.Api.Types.Exceptions;
+using Moq;
+using NUnit.Framework;
+using SFA.DAS.ExecutionPolicies;
+using SFA.DAS.NLog.Logger;
+using System;
+using System.Threading.Tasks;
+
+namespace SFA.DAS.UnitTests.ExecutionPolicies.HmrcExecutionPolicyTests
+{
+    [TestFixture]
+    public class PolicyRetryTest
+    {
+        private HmrcExecutionPolicy _policy;
+
+        [SetUp]
+        public void Init()
+        {
+            _policy = new HmrcExecutionPolicy(Mock.Of<ILog>(), TimeSpan.FromMilliseconds(0));
+        }
+
+        [Test]
+        public void LimitedRetryWhenServiceUnavailableReturned()
+        {
+            var runner = new TestRunner<ApiHttpException>(CreateTestException(503, "service unavailable"));
+
+            Assert.Throws<AggregateException>(() => _policy.ExecuteAsync(runner.Execute).Wait());
```

Comment:
Is it worth adding a timeout to the wait()? Since the behaviour of the retry logic is being tested it is possible that it might not work and just keep re-trying for ever, which would cause the test to hang. Adding a timeout would guard against this. 
I'm always suspicious of blocking calls. Could these be avoided by returning a task from the test and/or using throwsasync() may be?

Reply:
changes made